veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,225 @@
+- Enable namespace admins to perform NIC hotplug / unplug operations independently - without involving the cluster-admin 
+for each hotplug / hotunplug operation.
+
+## Non Goals
+
+## Definition of Users
+
+- Namespace administrators
+
+## User Stories
+
+As a namespace admin I would like to be able to:
+- Hotplug / unplug a secondary interface without involving the cluster admin.
+- Hotplug a new secondary network interface that uses bridge binding.
+- Hotunplug a secondary network interface that uses bridge binding.
+- Hotplug a new secondary network interface of type SR-IOV.
+- Use the staging rollout strategy so any hotplug / hotunplug operation will take effect only after restarting a VM.
+
+## Repos
+
+- kubevirt/kubevirt
+
+## Design
+
+### Integrate with LiveUpdate Rollout Strategy
+
+The CPU and memory hotplug features use the `LiveUpdate` [rollout strategy](https://kubevirt.io/user-guide/user_workloads/vm_rollout_strategies/)
+which automatically initiates a migration of the VM when certain spec fields are changed.
+
+Extend the feature to automatically initiate a live migration in case of NIC hotplug / hotunplug request.
+
+#### API Additions
+
+Currently, the CPU, memory and volume hotplug / hotunplug flows have a dedicated condition on the VMI object to mark the VMI
+for automatic migration:
+- HotVCPUChange
+- HotMemoryChange
+- VolumesChange
+
+Since we do not wish to introduce another resource-specific condition, a new generic allowed value will be added to `VirtualMachineInstanceConditionType`:
+
+```go
+VirtualMachineInstanceMigrationRequired VirtualMachineInstanceConditionType = "MigrationRequired"
+```
+
+Existing hotplug / hotunplug flows could be adjusted to use it.
+
+#### Migration Based Hotplug Flow
+
+1. The VM owner adds one or more new secondary vNICs and matching networks to a running VM.
+2. The VM controller patches the VMI object with the additional relevant vNICs and networks.
+3. The VMI controller:
+   
+   - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`) - since the dynamic-networks-controller is not present, this will not have an affect.
+   - Updates the VMI interface status.
+   - Adds the `MigrationRequired` condition to the VMI with its status set to `False` and a timestamp.
+4. On a future iteration of the reconciliation loop, the VMI controller will update the `MigrationRequired` condition status to `True` (Following no change in the interface statuses for more than TBD seconds after the initial timestamp set on the condition).
+5. The Workload update controller will identify the VMI as a candidate for automatic migration and will create a VMIM object for it.
+6. The Migration controller will remove the `MigrationRequired` condition once the migration is complete.
+7. The vNICs will be hot-plugged following the migration - same as today. 
+
+#### Migration Based Hotunplug Flow
```

Comment:
The logic is almost identical to the hotplug flow, unless I missed something.
That is a good thing that we want to emphasize, so it looks odd to see it separated here.
I see it was combined in the "In-Place" case.

Was there a special reason to have this separation?

Reply:
Thank you for the suggestion.

Done.